app/assets/javascripts/router.js

```diff
@@ -121,16 +121,49 @@ class ReactRouter extends React.Component<Props> {
                 path="/"
                 render={() =>
                   isAuthenticated ? (
-                    <DashboardView userId={null} isAdminView={false} />
+                    <DashboardView userId={null} isAdminView={false} initialTabKey={null} />
                   ) : (
                     <SpotlightView />
                   )
                 }
               />
+              <SecuredRoute
+                isAuthenticated={isAuthenticated}
+                path="/dashboard/:tab"
+                render={({ match }: ContextRouter) => {
+                  let initialTabKey: ?string;
+                  switch (match.params.tab) {
```

Comment:
The switch case here implements a mapping from url token to tab key which can be expressed more concisely using a simple object. Something like this should work:

```
const urlTokenToTabKeyMap = {
  gallery: "datasets",
  datasets: "advanced-datasets",
  tasks: "tasks",
  annotations: "explorativeAnnotations"
}
const initialTabKey = urlTokenToTabKeyMap[match.params.tab];
```

Reply:
The mapping object should probably be defined in the dashboard file (where the other variables are defined). If you need it in this file, you can just import it.